esp/src/src/react/components/ProfileManager.tsx

```diff
@@ -0,0 +1,159 @@
+import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
+import { ThemeEditor } from '@hpcc-js/other';
+
+interface ProfileManagerProps {
+    // username: string,
+    // accountType: string
+}
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+        root: {
+            '&:hover': {
+                background: "rgb(56, 79, 107)"
+            },
+            borderRadius: theme.shape.borderRadius
+        },
+        container: {
+            background: "transparent",
+            '&:hover': {
+                background: "transparent"
+            },
+            padding: theme.spacing(1,2),
+            flexDirection: 'row',
+            display: 'flex',
+            boxShadow: 'none'
+        },
+        flexCenter: {
+            minWidth: "200px",
+            display: "flex",
+            flexDirection: 'column',
+            alignItems: 'center'
+        },
+        avatar: {
+            width: theme.spacing(4),
+            height: theme.spacing(4),
+            backgroundColor: theme.palette.secondary.main,
+            background: "transparent",
+            padding: theme.spacing(1)
+        },
+        user: {
+            alignItems: 'center',
+            fontWeight: 'bold',
+            color: "white",
+            paddingLeft: theme.spacing(2),
+            padding: theme.spacing(0, 0, 1, 2)
+        },
+        role: {
+            color: theme.palette.primary.contrastText,
+            paddingLeft: theme.spacing(2)
+        },
+        text: {
+            textAlign: 'left'
+        },
+        arrow: {
+            color:theme.palette.common.white,
+            marginTop: '2px',
+            padding: theme.spacing(0, 2, 0, 2)
+        },
+        fullWidth: {
+            minWidth: "200px",
+            marginTop: theme.spacing(2)
+        },
+        userDetails: {
+            padding: theme.spacing(3, 0, 1, 0)
+        }
+  })
+);
+
+export const ProfileManager: React.FC<ProfileManagerProps> = (props) => {
+    const classes = useStyles();
+    const [open, setOpen] = React.useState(false);
+    const anchorRef = React.useRef<HTMLButtonElement>(null);
+    const prevOpen = React.useRef(open);
+
+    const handleToggle = () => {
+            setOpen(prevOpen => !prevOpen);
+    };
+
+    const handleClose = (event: React.MouseEvent<EventTarget>) => {
+        if (anchorRef.current && anchorRef.current.contains(event.target as HTMLElement)) {
+            return;
+        }
+        setOpen(false);
+    }
+
+    function handleListKeyDown(event: React.KeyboardEvent) {
+        if (event.key === 'Tab') {
+            event.preventDefault();
+            setOpen(false);
+        }
+    }
+
+    React.useEffect(() => {
+        if (prevOpen.current === true && open === false) {
+            anchorRef.current!.focus();
+        }
+
+        prevOpen.current = open;
+    }, [open]);
```

Comment:
See the "usePrevious" pattern here:  https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state

Reply:
Added new hook in hooks folder to cover this.